src/types.mo

```diff
@@ -7,213 +7,29 @@ import L "mo:stdlib/list";
 
 module {
 
-public module Sheet {
+// A generic Adapton engine is parameterized by these choices:
+public type EvalSig = module {
+  type Name;
+  nameEq : (n1:Name, n2:Name) -> Bool;
```

Comment:
@crusso I recall you saying that we cannot have type members in the module types, forbidding what I'm attempting to say above, right?

Reply:
Yup.

Reply:
It's just an unimplemented feature, not a fundamental restriction I hope.
For now, you could either parameterize EvalSig on the type componenents, assuming they aren't higher-kinded, or you could try to define EvalSig indirectly, using a module class, which should export the type components from the body. 
I'll try that on a small example to verify.

Reply:
Ah, the module class idea hadnt occured to me.  I don't see it yet, but I also haven't yet tried either.